### DISCUSSION TITLE
how to define a variable for frequency-dependnet surface impedance boundary?

### DISCUSSION THREAD

**Discussion by ZiadHatab** [ansys/pyaedt#5150] [Q&A] [answered]
Hi, I know that I can assign an impedance boundary with `assign_impedance_to_sheet()`, but how can I make a frequency dependent surface impedance? For example, I have a list of values for Zs as a function of frequency (same frequency grid as in sweep).

The online example is only showing a constant value
https://aedt.docs.pyansys.com/version/0.9/API/_autosummary/pyaedt.hfss.Hfss.assign_impedance_to_sheet.html 

Thanks!

**Comment by Samuelopez-ansys** [comment, marked as answer]
Hi @ZiadHatab,

I guess you know how to do it manually right?

Just in case, you first need to create a dataset with this information (Zs vs Freq). If the dataset has the name ds1, you just need:

`hfss.assign_impedance_to_sheet(rect.name, "TL2", "pwl(ds1, Freq)", 25)`

If you do not know how to create a dataset, here you can find an example:

```
x_datalist = [45, 53, 60, 70]
y_datalist = [0.5, 3, 6, 9]
hfss.create_dataset("PowerDissipationDataset", x_datalist, y_datalist, z=None, v=None, is_project_dataset=False,
                   x_unit="GHz", y_unit="ohm", v_unit="")
```

**Comment by ZiadHatab** [comment]
Cool, thanks!
